client/rpmtrans.c

```diff
@@ -480,12 +465,57 @@ TDNFTransAddInstallPkg(
                   fp,
                   pszFilePath,
                   &rpmHeader);
-    //If not checking gpg sigs, ignore signature errors
-    if(!nGPGSigCheck && (dwError == RPMRC_NOTTRUSTED || dwError == RPMRC_NOKEY))
+
+    Fclose(fp);
+    fp = NULL;
+
+    if (dwError != RPMRC_NOTTRUSTED && dwError != RPMRC_NOKEY)
     {
-        dwError = 0;
+        BAIL_ON_TDNF_RPM_ERROR(dwError);
+    }
+    else if(nGPGSigCheck)
```

Comment:
nit: No need of else here. just if will suffice

Reply:
Good point.